### PR TITLE
Send Bifrost verification error to Saturn

### PR DESCRIPTION
**Pull request by aarshkshah1992**

### CODE REVIEW

**Review by DiegoRBaquero — APPROVED** [review]

File: fetcher.go [hunk 1]
```diff
@@ -219,12 +220,13 @@ func (p *pool) fetchResource(ctx context.Context, from string, resource string,
...
-					NodeId:         saturnNodeId,
-					NodeIpAddress:  from,
-					IfNetworkError: networkError,
+					Range:             "",
+					Referrer:          respReq.Referer(),
+					UserAgent:         respReq.UserAgent(),
+					NodeId:            saturnNodeId,
+					NodeIpAddress:     from,
+					IfNetworkError:    networkError,
+					VerificationError: verificationError,
```
Comment (willscott) [inline fetcher.go:229]: how does this get set?  / how is this ever not ""

Comment (aarshkshah1992) [reply fetcher.go:229]: See line 360 below ?

File: fetcher.go [hunk 2]
```diff
@@ -352,7 +354,15 @@ func (p *pool) fetchResource(ctx context.Context, from string, resource string,
 	err = cb(resource, &wrapped)
 	received = wrapped.len
 	if err != nil {
-		goLogger.Errorw("failed to read response", "err", err.Error())
+
+		_, ok := err.(ErrInvalidResponse)
```
Comment (willscott) [inline fetcher.go:358]: can you do this as [errors.As](https://pkg.go.dev/errors#example-As)?

Comment (aarshkshah1992) [reply fetcher.go]: @willscott Done.

**Review by willscott — APPROVED** [review]

**Pull request merged**